runtime/win32.c

```diff
@@ -951,32 +951,64 @@ CAMLexport value caml_copy_string_of_utf16(const wchar_t *s)
   return v;
 }
 
-CAMLexport wchar_t* caml_stat_strdup_to_utf16(const char *s)
+Caml_inline wchar_t *strndup_noexc_to_utf16(const char *s,
+                                            size_t len, size_t *out_len)
 {
   wchar_t * ws;
   int retcode;
 
-  retcode = caml_win32_multi_byte_to_wide_char(s, -1, NULL, 0);
-  ws = caml_stat_alloc_noexc(retcode * sizeof(*ws));
-  caml_win32_multi_byte_to_wide_char(s, -1, ws, retcode);
+  retcode = caml_win32_multi_byte_to_wide_char(s, len, NULL, 0);
+  ws = caml_stat_alloc_noexc(retcode * sizeof(wchar_t));
```

Comment:
Same concern here as with `caml_stat_strndup`, you need to add `sizeof(wchar_t)` for the termination.